src/video/v9990/V9990.cc

```diff
@@ -338,7 +345,7 @@ void V9990::writeIO(word port, byte val, EmuTime::param time)
 			break;
 
 		case STATUS:
-			// read-only, ignore writes
+			getCliComm().printWarning("Writing to the V9990 read-only port ", portName[port]);
 			break;
```

Comment:
This comment applies both to this `printWarning()` and the one above.

This is fine for a first version. But I'm wondering if we shouldn't replace `printWarning()` with the `TclCallback` mechanism. In the past we indeed always used `printWarning()`, but many (all?) such warnings that trigger from emulation behavior (stuff that gets triggered by the emulated CPU) have been replaced by `TclCallback`. For example accessing VDP registers too fast, setting invalid PPI mode, excuting "DI ; HALT", ...

These callbacks can have a default implementation that prints the same warning message. But the main advantage is that power users can still replace this behavior. E.g. disable it, or log something, ...

Reply:
Should I put the function in `callbackprocs.tcl`? I am confused about the names of the callbacks. Some of them seem to use snake_case names (`invalid_psg_directions_callback`), some are squished together without separator (`vdpcmdinprogresscallback`). What name should I use? I think I will call them `v9990_invalid_port_read_callback` and `v9990_invalid_port_write_callback`.

Reply:
Yes, `share/scripts/callbackprocs.tcl` holds the stuff for our current callbacks. You should extend it for your new callbacks.

Unfortunately we haven't always been consistent with naming :(
But fixing the names isn't worth breaking backwards compatibility.
For new names we should try to do better. I like the style of the names you're proposing.

You've probably seen that the callbacks can take parameters. One parameter could be the port number. A second parameter could be an indication of "read or write". Though having separate callbacks for read and write is also fine (maybe it's even better .. I really haven't given it much thought yet .. I'll let you decide).

Yet another possibility is to allow for a future generic mechanism. I mean more generic than only V9990. Because e.g. also the V99x8 has write-only ports. Maybe pick generic names like `read_from_write_only_port_callback` and `write_to_read_only_port_callback` (no specific reference to V9990).

Note: I'm not expecting you to implement any of this future stuff.